### ISSUE TITLE
Question: how do I set timeout for mwoffliner?

### ISSUE THREAD

**Issue by tim-moody** [openzim/mwoffliner#1505]
I set --requestTimeout=120000 as an mwoffliner parameter and ran against mdwiki with a single page in the .tsv file. In the log I see:

```
[info] [2021-07-19T15:13:21.158Z] Worker [0] processing batch of article ids [["List_of_skin_conditions","WikiProjectMed:App/In>
[info] [2021-07-19T15:13:21.159Z] Getting article [List_of_skin_conditions] from https://mdwiki.org/w/api.php?action=visualedit>
[info] [2021-07-19T15:13:21.159Z] Getting JSON from [https://mdwiki.org/w/api.php?action=visualeditor&mobileformat=html&format=>
[error] [2021-07-19T15:14:16.388Z] Error in retrieved article [List_of_skin_conditions]: {
  code: ^[[32m'apierror-visualeditor-docserver-http-error'^[[39m,
  info: ^[[32m'Error contacting the Parsoid/RESTBase server: (curl error: 28) Timeout was reached'^[[39m,
  ^[[32m'*'^[[39m: ^[[32m'See https://mdwiki.org/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list >
}
[warn] [2021-07-19T15:14:16.391Z] No HTML returned for article [List_of_skin_conditions], skipping

```

looks like it timed out after 55 seconds

**Comment by kelson42** [comment]
This is something I would like to investigate a bit as well... and change the timeout from ms to s.

**Comment by kelson42** [comment]
@MananJethwani Maybe you have answer to that?

**Comment by MananJethwani** [comment]
@kelson42 sorry for the late reply, I will look into it later today and let you know about the exact issue.

**Comment by kelson42** [comment]
@tim-moody Unfortunately we can not reproduce this behaviour so far.

**Comment by MananJethwani** [comment]
@kelson42 I confirm the timeout is respected and there is no such issue,
@tim-moody one thing I want to mention is the timeout here doesn't mean connection timeout instead it means response timeout.
may be you confused it with that.

**Comment by kelson42** [comment]
I think we should close this ticket as there is no reprocase successful. But the timeout problem stays the same with mdwiki, see https://farm.openzim.org/pipeline/8c164e96f8d15f6cacf28f06/debug. On our side we have increased the default timeout from 60s to 120s to help to accomodate, but this seems not enough.

**Comment by tim-moody** [comment]
From @MananJethwani it seems there are two timeouts and only one is settable. The first is called 'connection' timeout, but seems to wait for first results, whereas the second, which is settable, is total duration of the time to complete the reply. I suggest you either add a setting for 'connection' timeout or set it to the same value as the existing timeout.

I think you would see this problem with any sufficiently slow mediawiki, but the only one experienced is mdwiki, and it is always slow on a couple of articles whose 'connection' always times out unless they are in the mdwiki cache.

**Comment by kelson42** [comment]
@MananJethwani what is the connection timeout? What is the value of the connection timout? i don’t inderstand.

**Comment by tim-moody** [comment]
I have asked the maintainer of the mdwiki mediawiki to comment on the configuration lines which I just discovered.

// InstantCommons causes some very slow parses; we don't want those to time out
$wgHTTPTimeout = 55;
$wgAsyncHTTPTimeout = 55; 

@MananJethwani if the mdwiki server terminated the connection in 55 seconds, would that to what you have called connection timeout?

**Comment by kelson42** [comment]
@tim-moody, @rgaudin has discovered a timeout here https://github.com/offspot/kiwix-hotspot/commit/92389370fb3c4beaf208c913e0d0597180f0798a. Might be useful for this ticket.

**Comment by tim-moody** [comment]
I think the question has been answered and this ticket can be closed.

**Comment by tim-moody** [comment]
When I opened this ticket I assumed that mwoffliner was timing out on some api requests as I observed mdwiki to be very slow in returning large pages and even at times failing to do so. I now believe that what happens is that if mdwiki times out in responding it returns either a 503 status or a status 200 with json describing the error. Either of these responses will cause mwoffliner to error out and stop.

It is possible to retry both of these errors, however, and eventually mdwiki returns a valid response. This situation is most often seen with the api request.

`/w/api.php?action=visualeditor&mobileformat=html&format=json&paction=parse&page=`

I don't know if this has general applicability beyond mdwiki.
